crates/nu-parser/src/lex.rs

```diff
@@ -299,13 +299,26 @@ pub fn lex_item(
     }
 }
 
+// borrowed from rustc-lexer
+fn strip_shebang(input: &[u8]) -> &[u8] {
+    let src = String::from_utf8_lossy(input);
+    if let Some(rest) = src.strip_prefix("#!") {
+        // (the `#`) + (the '!') = 2
+        let len = 2 + rest.lines().next().unwrap_or_default().len();
+        &input[len..]
+    } else {
+        input
+    }
+}
+
 pub fn lex(
     input: &[u8],
     span_offset: usize,
     additional_whitespace: &[u8],
     special_tokens: &[u8],
     skip_comment: bool,
 ) -> (Vec<Token>, Option<ParseError>) {
+    let input = strip_shebang(input);
```

Comment:
Mhh can this mess with what ever `span_offset` means in this coontext?

I am not sure if we properly test the correct relative position of error spans in enough tests? Most functionality tests ignore the spans as their exact absolute position is subject to frequent change or not determinate.

Reply:
Imagine you `source` a file that has a shebang. Wouldn't the source code that is indexed and kept around have shifted span positions due to the skipping of the shebang?

Reply:
That's a good point, we need to make sure about it. The `span_offset` points at the end of all the source processed by the parser so far. We're appending new source on top of it, so it should be fine. However, during lexing, the full source including shebang has already been added to the engine state: https://github.com/nushell/nushell/blob/3fb1e37473ce998b223844f2ae0275a4ebe3672f/crates/nu-parser/src/parser.rs#L5999. This might lead to span mismatch weirdness. The shebang should then be stripped even before that line.